include/open62541pp/types/Variant.h

```diff
@@ -72,54 +69,35 @@ class Variant : public TypeWrapper<UA_Variant, UA_TYPES_VARIANT> {
         return variant;
     }
 
-    /// Create Variant from array (no copy if assignable without conversion).
-    template <typename T>
-    [[nodiscard]] static Variant fromArray(Span<T> array) {
+    /// Create Variant from array (no copy if assignable).
+    template <typename ArrayLike>
+    [[nodiscard]] static Variant fromArray(ArrayLike&& array) {
+        using ValueType = typename std::remove_reference_t<ArrayLike>::value_type;
+        constexpr bool isMutable = detail::IsMutableContainer<ArrayLike>::value;
+        constexpr bool isContiguous = detail::IsContiguousContainer<ArrayLike>::value;
         Variant variant;
-        if constexpr (detail::isRegisteredType<T>) {
-            variant.setArray(array);
+        if constexpr (!std::is_rvalue_reference_v<decltype(array)> && isMutable && isContiguous && detail::isRegisteredType<ValueType>) {
```

Comment:
Looks good. 
Just one last thing: 

```C++
std::array<int, 3> array{1, 2, 3}
Variant::fromArray(Span(array));
```
This does now copy the array, even if it is safe to just set it. 
But I don't think there is a nice solution for this without an additional overload for Span.

Reply:
The only option I see (without adding overloads) is to go back to universal references and static_assert, that the object is not a temporary container: https://github.com/open62541pp/open62541pp/pull/174/commits/385c9f0da7eac00c1d55a7db5e2ecc1aa9f0e91a

Improvement or not?

Reply:
looks good 👍